src/ripple/nodestore/impl/Shard.h

```diff
@@ -56,13 +56,17 @@ class Shard final
         Application& app,
         DatabaseShard const& db,
         std::uint32_t index,
-        beast::Journal j);
+        beast::Journal j,
+        boost::filesystem::path const& dir = "");
```

Comment:
`journal` are usually last parameter. In this, default parameters won't work with that. We'll either two functions or we can keep `journal` where it is. Your call.

Reply:
Added a second constructor